packages/account/src/predicate/predicate.ts

```diff
@@ -43,12 +43,14 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
    * @param bytes - The bytes of the predicate.
    * @param provider - The provider used to interact with the blockchain.
    * @param jsonAbi - The JSON ABI of the predicate.
+   * @param predicateData - The predicate data.
    * @param configurableConstants - Optional configurable constants for the predicate.
    */
   constructor(
     bytes: BytesLike,
     provider: Provider,
     jsonAbi?: JsonAbi,
+    predicateData?: ARGS,
     configurableConstants?: { [name: string]: unknown }
```

Comment:
As we are introducing a breaking change here _anyway_, and we've now got 5 params, is it worth us using an object for the constructor predicate params. This _may_ defend against future breaking changes with the predicate constructor.

```ts
constructor(params: PredicateParams)
const { bytes, provider, jsonAbi, predicateData, configurableConstants } = params;
```

I appreciate this is a bigger breaking change, and your change would not impact anyone instantiating via the required params, but thought I'd suggest as we are now at 5 params.

Reply:
Fair point. The breaking change would be more significant, but the API would be more straightforward.

It's a delicate one; we should think this through.

Reply:
If we are going to go down that route then I'd suggest also removing the `setData` method. It'd break everybody's `Predicate` usage anyways, so it makes sense to introduce another breaking change that's in the same vein.

Reply:
@danielbate TBH I'm in favour of using objects for the constructor params too. Judging from the sync, it looks like this is the consensus between the team, so I'll go ahead and incorporate this change.

Reply:
@nedsalk I'm sorry but I don't follow. You would still need `setData` if you want to pass in some different data after instantiating the predicate. I don't mind the breaking change, but I feel like this method is needed.

If we remove this method, the only way to pass in different data to a predicate would be to re-instantiate it with different data. I don't think that DX would be great. 

I personally think of a `Predicate` instance as a 'pointer' to a predicate. Kinda like how a `Contract` instance is a pointer to a contract. You can call different functions on the contract once you have the instance, and you don't re-instantiate it for every function call.

For predicates, the different function calls are just different inputs:
<img width="911" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/39617427/5cb9a3da-9333-4bc9-94dc-0a087f897adb">

Reply:
@Dhaiwat10 I mentioned it because we kinda had an agreement with the frontend team in one of our syncs that it makes sense to just have it in the constructor and remove `setData`.

The framing you presented is valid. As far as I remember, the logic behind the suggestion was that `predicateData` is mostly seen as an equivalent to a private key for a walletn:
- If you don't have the private key that unlocks the wallet, you can't instantiate the unlocked wallet
- If you don't have the data that unlocks the predicate, there's no point in instantiating the predicate itself

The benefit of not having `setData` from the internal maintenance standpoint is that state management is easier when we know that the data is set _only_ once. Just like we don't allow the changing of a `BaseWalletUnlocked`'s private key but rather you have to instantiate a new `BaseWalletUnlocked`, we wouldn't allow the changing of predicate data.

Reply:
Sounds fair @nedsalk. If this has already been discussed with the frontend team, then I will remove it